pkg/controller/autoops/expected.go

```diff
@@ -0,0 +1,276 @@
+			},
+			Privileged: ptr.To(false),
+			// Can't set this to true because of:
+			// failed to build pipelines:
+			// failed to create "metricbeatreceiver" receiver for data type "logs":
+			// error creating metricbeatreceiver: error loading meta data:
+			// failed to create Beat meta file: open data/meta.json.new: read-only file system
+			ReadOnlyRootFilesystem: ptr.To(false),
+			// Can't currently do this because of:
+			// Error: container has runAsNonRoot and image has non-numeric user (elastic-agent)
+			// RunAsNonRoot:           ptr.To(true),
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+		}).
+		PodTemplate
+
+	return common_deployment.New(common_deployment.Params{
+		Name:      autoopsv1alpha1.Deployment(policy.GetName(), es),
+		Namespace: policy.GetNamespace(),
+		Selector: map[string]string{
+			autoOpsLabelName: policy.GetName(),
+		},
+		Metadata:             meta,
+		PodTemplateSpec:      podTemplateSpec,
+		Replicas:             1,
+		RevisionHistoryLimit: policy.Spec.RevisionHistoryLimit,
+	}), nil
+}
+
+// readinessProbe is the readiness probe for the AutoOps Agent container
+func readinessProbe() corev1.Probe {
+	scheme := corev1.URISchemeHTTP
+	return corev1.Probe{
+		FailureThreshold:    3,
+		InitialDelaySeconds: 5,
+		PeriodSeconds:       10,
+		SuccessThreshold:    1,
+		TimeoutSeconds:      5,
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Port:   intstr.FromInt(readinessProbePort),
+				Path:   "/health/status",
+				Scheme: scheme,
+			},
+		},
+	}
+}
+
+// buildConfigHash builds a hash of the ConfigMap data and secret values
+// to trigger pod restart on config changes
+func buildConfigHash(ctx context.Context, c k8s.Client, policy autoopsv1alpha1.AutoOpsAgentPolicy, es esv1.Elasticsearch) (string, error) {
+	configHash := fnv.New32a()
+
+	// Hash ConfigMap data
+	configMapName := autoopsv1alpha1.Config(policy.GetName(), es)
+	var configMap corev1.ConfigMap
+	configMapKey := types.NamespacedName{Namespace: policy.Namespace, Name: configMapName}
+	if err := c.Get(ctx, configMapKey, &configMap); err != nil {
+		return "", err
+	}
+
+	if configData, ok := configMap.Data[autoOpsESConfigFileName]; ok {
+		_, _ = configHash.Write([]byte(configData))
+	}
+
+	// Hash secret values from autoops-secret
+	autoopsSecretKey := types.NamespacedName{Namespace: policy.Namespace, Name: "autoops-secret"}
+	var autoopsSecret corev1.Secret
+	if err := c.Get(ctx, autoopsSecretKey, &autoopsSecret); err != nil {
+		return "", fmt.Errorf("failed to get autoops-secret: %w", err)
+	}
+
+	// Hash secret keys, including optional keys. There's no code here to handle missing keys as
+	// 1. The optional keys are included here.
+	// 2. The required keys are already validated in the controller, so they should always be present.
+	requiredKeys := []string{"autoops-token", "autoops-otel-url", "cloud-connected-mode-api-key", "cloud-connected-mode-api-url"}
```

Comment:
Why are awe not using the constants defined in configuration.go? Why is there an extra `cloud-connected-mode-api-url` I remember that some environments have this? This deserves at least a comment?

Reply:
Just an oversight. I've updated them to use constants, and added a comment concerning the optional field. Lmk if it's good.